plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,572 @@
+                        .build());
+    }
+
+    @Override
+    public Collection<String> listSchemas(Connection connection)
+    {
+        return ImmutableList.of(IGNITE_SYS_SCHEMA, IGNITE_SCHEMA);
+    }
+
+    @Override
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(connection.getCatalog(),
+                connection.getSchema(),
+                tableName.orElse(null),
+                null);
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        // TODO complete type mapping
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+                return Optional.of(varcharColumnMapping(
+                        typeHandle.getColumnSize()
+                                // default added column in Ignite will has Integer.MAX_VALUE length which not allow in Trino, will consider the column as varchar
+                                .filter(e -> e != Integer.MAX_VALUE)
+                                .map(VarcharType::createVarcharType)
+                                .orElse(createUnboundedVarcharType()), true));
+            case Types.DATE:
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        dateReadFunction(),
+                        dateWriteFunction()));
+        }
+        return legacyColumnMapping(session, connection, typeHandle);
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        return aggregateFunctionRewriter.rewrite(session, aggregate, assignments);
+    }
+
+    private static Optional<JdbcTypeHandle> toTypeHandle(DecimalType decimalType)
+    {
+        return Optional.of(new JdbcTypeHandle(Types.DECIMAL, Optional.of("Decimal"), Optional.of(decimalType.getPrecision()), Optional.of(decimalType.getScale()), Optional.empty(), Optional.empty()));
+    }
+
+    // https://issues.apache.org/jira/browse/IGNITE-12824, user do not need to convert between UTC <-> Local time
+    public static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            long localMillis = resultSet.getDate(columnIndex).getTime();
+            return MILLISECONDS.toDays(localMillis);
+        };
+    }
+
+    public static LongWriteFunction dateWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long millis = DAYS.toMillis(value);
+            statement.setDate(index, new Date(millis));
+        };
+    }
+
+    @Override
+    public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHandle tableHandle)
```

Comment:
Why this method is inlined? What are the differences between trino-base-jdbc? Can we please avoid the copy paste?

Reply:
Removed